packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -1,23 +1,40 @@
 import cxapi = require('@aws-cdk/cx-api');
+import { makeUniqueId } from '../util/uniqueid';
 export const PATH_SEP = '/';
 
 /**
  * Represents the building block of the construct graph.
  * When a construct is created, it is always added as a child
  */
 export class Construct {
-    private static readonly VALID_NAME_REGEX = /^[A-Za-z][A-Za-z0-9]*$/;
-
     /**
      * Returns the parent of this node or undefined if this is a root node.
      */
     public readonly parent?: Construct;
 
     /**
-     * The name of this construct
+     * @deprecated use `id`
```

Comment:
Yeahhhhhh that's a poor deprecation message. You want to explain why `id` is better.

Reply:
Also a good idea to mention *when* it was deprecated, so we can assess *when* it is safe to drop.

Reply:
Finally, why are you so careful in not breaking usage of `.name`? I feel like this is probably not common place... Would have dropped.

Reply:
okay, removed!